python/ffsim/states/slater.py

```diff
@@ -0,0 +1,353 @@
+            sampled_configuration_b = np.ones(
+                (n_chains * (chain_length + 1), norb), dtype=int
+            ) * int(n_a / norb)
+        else:
+            sampled_configuration_b = _sample_spinless(
+                rdm_b, chain_length, n_chains, n_particles_to_move, seed
+            )
+
+        sampled_configuration = np.concatenate(
+            (sampled_configuration_b, sampled_configuration_a), axis=1
+        )
+
+    return sampled_configuration
+
+
+def _propose_move(
+    previous_step: np.ndarray, norb: int, n_particles_to_move: int, rng=None
+) -> np.ndarray:
+    """
+    Proposes a new state in the Markov chain. The new state is obtained by allowing
+    at most ``n_particles_to_move`` to change the orbital that they occupy.
+    The new orbitals that they can occupy are chosen at random from the set of
+    unoccupied orbitals.
+
+    Args:
+        previous step: A Numpy array with the starting positions of the electrons.
+        norb: Number of orbitals.
+        n_particles_to_move: At each step of the chain, the maximum number of
+            electrons that are moved into a different orbital.
+
+    Returns:
+        A Numpy array with the proposed new set of positions of the electrons.
+    """
+
+    rng = np.random.default_rng(rng)
+
+    n_chains = previous_step.shape[0]
+    nelec = previous_step.shape[1]
+
+    positions = np.arange(norb)
+
+    new_state = np.zeros((n_chains, nelec), dtype=int)
+
+    for i in range(n_chains):
+        sample_particles_to_move = rng.choice(np.arange(n_particles_to_move) + 1, 1)
+
+        open_orbitals = np.setdiff1d(positions, previous_step[i], assume_unique=True)
+
+        electron_id = np.sort(
+            rng.choice(np.arange(nelec), sample_particles_to_move, replace=False)
+        )
+
+        new_positions = rng.choice(
+            open_orbitals, sample_particles_to_move, replace=False
+        )
+
+        new_state[i] = previous_step[i]
+        new_state[i, electron_id] = new_positions
+        new_state[i] = np.sort(new_state[i])
+
+    return new_state
+
+
+def _evaluate_logdeterminant_squared(
+    positions: np.ndarray, phi: np.ndarray
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Given a batch of positions of electronic configurations, evaluate their
+    Slater determinant.
+
+    Args:
+        positions: A 2D Numpy array of electronic configurations. Each row represents
+            the position of the electrons for each configuration.
+        phi: A 2D (norb, nelec) Numpy array. The matrix must be column-wise
+            orthogonal. Each column represents an orbital.
+
+    Returns: Tuple. The first element is a Numpy array with the signs of the
+        determinants. The second element is a Numpy array with the logarithm
+        of the absolute value of the determinant.
+    """
+
+    det_matrices = phi[positions, :]
+
+    signs, logdet_amps = np.linalg.slogdet(det_matrices)
+
+    return signs, logdet_amps
```

Comment:
This helper function seems unnecessary. Remove it and simply call `np.linalg.slogdet` directly instead.